js/app/modules/navigation/topMenu.js

```diff
@@ -73,42 +78,14 @@ const TopMenu = new Module.Class({
     },
 
     vfunc_size_allocate: function (alloc) {
-        this.parent(alloc);
+        let [banner_min,] = this._banner.get_preferred_width();
+        let [menu_min,] = this._menu.get_preferred_width();
 
-        let [, banner_nat_width] = this._banner.get_preferred_width();
-        let [, menu_nat_width] = this._menu.get_preferred_width();
-        let [, banner_nat_height] = this._banner.get_preferred_height();
-        let [, menu_nat_height] = this._menu.get_preferred_height();
-
-        let show_banner = (alloc.width >= banner_nat_width + menu_nat_width);
-
-        let x = alloc.x;
-        if (show_banner) {
-            let banner_rect = new Gdk.Rectangle({
-                x: x,
-                y: alloc.y + _get_centered_coord(alloc.height, banner_nat_height),
-                width: banner_nat_width,
-                height: banner_nat_height,
-            });
-            this._banner.size_allocate(banner_rect);
-            x += alloc.width - menu_nat_width;
-        } else {
-            x += Math.max(0, (alloc.width - menu_nat_width) / 2);
-        }
-        this._banner.set_child_visible(show_banner);
-
-        let menu_rect = new Gdk.Rectangle({
-            x: x,
-            y: alloc.y + _get_centered_coord(alloc.height, menu_nat_height),
-            width: Math.min(alloc.width, menu_nat_width),
-            height: menu_nat_height,
-        });
-        this._menu.size_allocate(menu_rect);
+        this._banner.visible = (alloc.width >= banner_min + menu_min);
```

Comment:
I think this needs to use `set_child_visible()` so that `show_all()` doesn't affect it.